src/dlgPackageExporter.cpp

```diff
@@ -268,64 +317,230 @@ void dlgPackageExporter::slot_export_package()
+                        break;
+                    }
+                }
+            }
+
+            if (isOk) {
+                if (fileEntries.contains(xmlFileName) && fileEntries.value(xmlFileName) == mXmlPathFileName) {
+                    if (!writeFileToZip(xmlFileName, mXmlPathFileName, archive)) {
+                        isOk = false;
+                    }
+
+                    // If successful will get to HERE...
+
+                } else {
+                    displayResultMessage(tr("Required \"%1\" file that contains the mudlet items chosen for the "
+                                            "module was not found in the staging area from where it would be include in the "
+                                            "module (archive) file - this suggests there is a problem with the "
+                                            "permissions for the \"%2\" directory!")
+                                            .arg(mXmlPathFileName, QDir(mStagingDirName).canonicalPath()), false);
+                    isOk = false;
+                }
+            }
+
+            if (isOk) {
+                // THIS is the point that the archive gets created from the
+                // source materials - it may take a short while!
+                // If it fails to write out the new file 'archive' is left
+                // unchanged (and we can still access it to get the error
+                // details):
+                ze = zip_close(archive);
+                if (ze) {
+                    displayResultMessage(tr("Failed to populate and then close module (archive) file, error is: \"%1\".")
+                                         .arg(QString::fromUtf8(zip_strerror(archive))), false);
+                    // In libzip 0.11 a function was added to clean up
+                    // (deallocate) the memory associated with an archive
+                    // - which would normally occur upon a successful close
+                    // - before that version the memory just leaked away...
+                    zip_discard(archive);
+                    isOk = false;
+                }
+
+            } else {
+                zip_discard(archive);
+            }
         }
     }
-    err = zip_close(archive);
-    if (err != 0) {
-        zip_error_to_str(buf, sizeof(buf), err, errno);
-        //FIXME: report error to userqDebug()<<"dp close file error"<<buf;
-        close();
-        return;
-    }
-
-    showUploadNudge();
-
-}
 
-void dlgPackageExporter::showUploadNudge()
-{
-   ui->infoLabel->setText(R"(<a href="https://forums.mudlet.org/posting.php?mode=post&f=6">Upload package to Mudlet</a>)");
-   ui->infoLabel->setTextFormat(Qt::RichText);
-   ui->infoLabel->setTextInteractionFlags(Qt::TextBrowserInteraction);
-   ui->infoLabel->setOpenExternalLinks(true);
+    if (isOk) {
+        // Success!
+        displayResultMessage(tr("Exported package, it is at: \"%1\".").arg(mPackagePathFileName), true);
```

Comment:
How about `Package exported to:` or `Package saved at:`

Reply:
Yeah, I'll take the first of those suggestions...

Reply:
Done.